cpp_api/objects/equation.cpp

```diff
@@ -162,12 +162,13 @@ char Equation::get_solved() const
 
 int Equation::get_method_as_int() const
 {
-    return (int) this->method;
+    int m = (int) this->method;
+    return (m >= 0 && m < I_NB_EQ_METHODS) ? m : 0;
```

Comment:
why is this test needed? how can this ever happen?

Reply:
Because in some old versions of IODE, when the memory space was allocated for a new equation (*EQ* struct), no default value was set for the *method* attribute of the *EQ* struct. In other words, eq->method contained *garbage*. 
The problem is that some users still use binary files in which equations are stored with *garbage* value for the *equation method*.

For exmaple, the *fun* model (with its *fun.xxx* binary files) is used as a `test` model and is used in all C and C++ test functions.
It is a real small econometric model written decades ago by I don't know who but not Jean-Marc.
The *fun.eqs* binary file contains equations having "garbage" value for their *method* attibute.

One the purpose of the C++ API was to serve as a kind of *firewall* between the old C API and the new Qt user interface.

Reply:
ouch